### ISSUE TITLE
Deploy NoClientDeploy fails on "DeployAD" resource

### ISSUE THREAD

**Issue by big-kahuna-burger** [Azure-Samples/active-directory-lab-hybrid-adfs#3]
I was able to run this `NoClientDeploy` setup before, but the recent deployment I've tried fails on resource with display name "DeployAD".

![depfail](https://user-images.githubusercontent.com/11753867/44955951-3ae57100-aebc-11e8-90b1-71cad59481c9.png)

Any help is appreciated.

**Comment by big-kahuna-burger** [comment]
From Activity Log:

```
"properties": {
        "statusCode": "Conflict",
        "statusMessage": "{\"status\":\"Failed\",\"error\":{\"code\":\"ResourceDeploymentFailure\",\"message\":\"The resource operation completed with terminal provisioning state 'Failed'.\",\"details\":[{\"code\":\"DeploymentFailed\",\"message\":\"At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-debug for usage details.\",\"details\":[{\"code\":\"Conflict\",\"message\":\"{\\r\\n  \\\"status\\\": \\\"Failed\\\",\\r\\n  \\\"error\\\": {\\r\\n    \\\"code\\\": \\\"ResourceDeploymentFailure\\\",\\r\\n    \\\"message\\\": \\\"The resource operation completed with terminal provisioning state 'Failed'.\\\",\\r\\n    \\\"details\\\": [\\r\\n      {\\r\\n        \\\"code\\\": \\\"VMExtensionProvisioningError\\\",\\r\\n        \\\"message\\\": \\\"VM has reported a failure when processing extension 'DeployAD'. Error message: \\\\\\\"Finished executing command\\\\\\\".\\\"\\r\\n      }\\r\\n    ]\\r\\n  }\\r\\n}\"}]}]}}",
        "serviceRequestId": null
    }
```

**Comment by big-kahuna-burger** [comment]
I've managed to mangle the `deploy.ps1` and to provision everything needed with posh. The reason I'm suspecting it won't work with template button and portal is that from `Microsoft.Compute` `CustomScriptExtension` class has a `TypeHandlerVersion` 1.4 and `autoUpdateMinorVersion` params set to `false`. Will check with my fork and let you know if that is the reason. In case it is I can send a PR to fix it.

I've used `Get-AzureRmVMExtensionImage -Location centralus -PublisherName Microsoft.Compute -Type CustomScriptExtension | Sort Version | select PublisherName,Type,Version` to get the versions history.
